### PR TITLE
added package ocamlsdl2

### PR DESCRIPTION
**Pull request by fccm**
added package ocamlsdl2 versions v0.02 and v0.03

### CODE REVIEW

**Comment by mseri** [comment]
Is the failure here due to an older or newer version of SDL?
```

#=== ERROR while compiling ocamlsdl2.0.03 =====================================#
# context              2.0.3 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/fccm/OCamlSDL2/archive/v0.03.tar.gz)
# path                 ~/.opam/4.07/.opam-switch/build/ocamlsdl2.0.03
# command              /usr/bin/make -C src opt
# exit-code            2
# env-file             ~/.opam/log/ocamlsdl2-20-50de41.env
# output-file          ~/.opam/log/ocamlsdl2-20-50de41.out
### output ###
# make: Entering directory '/home/opam/.opam/4.07/.opam-switch/build/ocamlsdl2.0.03/src'
# ocamlc -g sdlinit.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlinit.ml
# ocamlc -g sdlquit.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlquit.ml
# ocamlc -g sdlrect.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlrect.ml
# ocamlc -g sdlblendMode.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlblendMode.ml
# ocamlc -g sdlpixelFormat.mli
# ocamlc -g sdlsurface.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlpixelFormat.ml
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlsurface.ml
# ocamlc -g sdlwindow.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlwindow.ml
# ocamlc -g sdltype.mli
# ocamlc -g sdltexture.mli
# ocamlc -g sdlrender.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdltype.ml
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdltexture.ml
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlrender.ml
# ocamlc -g sdlpixel.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlpixel.ml
# ocamlc -g sdlscancode.mli
# ocamlc -g sdlkeymod.mli
# ocamlc -g sdlkeycode.mli
# ocamlc -g sdlhat.mli
# ocamlc -g sdlevent.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlscancode.ml
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlkeymod.ml
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlkeycode.ml
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlhat.ml
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlevent.ml
# ocamlc -g sdlkeyboard.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlkeyboard.ml
# ocamlc -g sdlmouse.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlmouse.ml
# ocamlc -g sdljoystick.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdljoystick.ml
# ocamlc -g sdlclipboard.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlclipboard.ml
# ocamlc -g sdltimer.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdltimer.ml
# ocamlc -g sdlaudio.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlaudio.ml
# ocamlc -g sdlfilesystem.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlfilesystem.ml
# ocamlc -g sdlversion.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlversion.ml
# ocamlc -g sdlcpuinfo.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlcpuinfo.ml
# ocamlc -g sdlpower.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlpower.ml
# ocamlc -g sdlrwops.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlrwops.ml
# ocamlc -g sdlerror.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlerror.ml
# ocamlc -g sdlgl.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdlgl.ml
# cat sdl_header > sdl.mli
# ocamlc -g -i sdl.ml >> sdl.mli
# ocamlc -g sdl.mli
# ocamlopt -g -cc "cc" -g -w A -warn-error A -ccopt "" -c sdl.ml
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlinit_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlquit_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlrect_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlblendMode_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlsurface_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlwindow_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlrender_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdltexture_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlpixelFormat_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlpixel_stub.c
# ocamlc -g -ccopt "-static -I/usr/include/SDL2 -D_REENTRANT -g -O " sdlevent_stub.c
# sdlevent_stub.c: In function 'Val_SDL_Event':
# sdlevent_stub.c:511:10: error: 'SDL_DISPLAYEVENT' undeclared (first use in this function)
#      case SDL_DISPLAYEVENT:               return Val_SDL_DISPLAYEVENT;
#           ^~~~~~~~~~~~~~~~
# sdlevent_stub.c:511:10: note: each undeclared identifier is reported only once for each function it appears in
# sdlevent_stub.c:518:10: error: 'SDL_SENSORUPDATE' undeclared (first use in this function)
#      case SDL_SENSORUPDATE:               return Val_SDL_SENSORUPDATE;
#           ^~~~~~~~~~~~~~~~
# Makefile:193: recipe for target 'sdlevent_stub.o' failed
# make: *** [sdlevent_stub.o] Error 2
# make: Leaving directory '/home/opam/.opam/4.07/.opam-switch/build/ocamlsdl2.0.03/src'
```

File: packages/ocamlsdl2/ocamlsdl2.0.02/opam [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  [make "-C" "src" "gen"]
+  [make "-C" "src" "dep"]
+  [make "-C" "src" "opt"]
+  [make "-C" "src" "byte"]
+]
+install: [
+  [make "-C" "src" "findinstall"]
+  [make "-C" "src" "findinstall_h"]
+]
+remove: [
```
Comment (mseri) [inline packages/ocamlsdl2/ocamlsdl2.0.02/opam]: Note that the remove fields are no longer necessary in opam2 when you only need to remove what you have installed

File: packages/ocamlsdl2/ocamlsdl2.0.02/opam [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+tags: [
+  "bindings" "graphics" "audio" "multimedia" "opengl"
+  "cross-platform"
+]
+synopsis: "Interface to the SDL2 library"
+description: """
+An OCaml interface to the SDL2 multimedia library for Linux and Windows."""
+depends: [
+  "ocaml"
+  "ocamlfind"
```
Comment (mseri) [inline packages/ocamlsdl2/ocamlsdl2.0.02/opam]: From a brief look at the code, it seems to me that `ocamlfind` is only used as `{build}` dependency

**Comment by fccm** [comment]
Yes, these enum values appeared in latest versions of the SDL2.
You can see from this commit that I added it recently:
https://github.com/fccm/OCamlSDL2/commit/4e63c68d6b6e8d85c829bdca708062fa24676bfc
This is why I commited to opam-repo 2 versions of ocamlsdl2: v0.02 and v0.03
You can see that version 0.02 compiles fine because you probably have an old version of the SDL2 in your system. v0.03 compiles fine with SDL 2.0.9 but I don't know how to constrain this in the opam file. There is probably similar problems with "tsdl" and the changes in the SDL2 API. Tsdl README says its made for SDL 2.0.6, have a look at this diff:
diff -Naur SDL2-2.0.6/include/SDL_events.h SDL2-2.0.9/include/SDL_events.h
https://gist.github.com/fccm/a7a2739137b99ad75e271f04cae95a60


ocamlfind is only used to install the files, you can check the makefile:
https://github.com/fccm/OCamlSDL2/blob/master/src/Makefile

**Comment by kit-ty-kate** [comment]
Looks good enough. Thanks!

**Pull request merged**